packages/enzyme/src/ShallowWrapper.js

```diff
@@ -509,17 +577,28 @@ class ShallowWrapper {
 
         // When shouldComponentUpdate returns false we shouldn't call componentDidUpdate.
         // so we spy shouldComponentUpdate to get the result.
-        let spy;
+        let shouldComponentUpdateSpy;
+        let getChildContextSpy;
         let shouldRender = true;
         if (
           !this[OPTIONS].disableLifecycleMethods
-          && lifecycles.componentDidUpdate
-          && lifecycles.componentDidUpdate.onSetState
           && instance
-          && typeof instance.shouldComponentUpdate === 'function'
         ) {
-          spy = spyMethod(instance, 'shouldComponentUpdate');
-        } else if (isPureComponent(instance)) {
+          if (
+            lifecycles.componentDidUpdate
+            && lifecycles.componentDidUpdate.onSetState
+            && typeof instance.shouldComponentUpdate === 'function'
+          ) {
+            shouldComponentUpdateSpy = spyMethod(instance, 'shouldComponentUpdate');
+          }
+          if (
+            lifecycles.getChildContext.calledByRenderer
+            && typeof instance.getChildContext === 'function'
+          ) {
+            getChildContextSpy = spyMethod(instance, 'getChildContext');
+          }
+        }
+        if (!shouldComponentUpdateSpy && isPureComponent(instance)) {
```

Comment:
I believe PureComponents completely ignore any shouldComponent methods, so if it's a PureComponent, it should always fall into this branch

Reply:
Interesting! Based on my reading of the [current implementation in master](https://github.com/airbnb/enzyme/blob/c7282ebc52aafd3442f7b52dde75354bae973674/packages/enzyme/src/ShallowWrapper.js#L379-L392), we don't do the `isPureComponent` check if `instance.shouldComponentUpdate` is a function (as per the first `if ()` condition.

So this implementation more closely matches the implementation in master. However, perhaps what's in master is wrong. In that case, would you like me to fix it here, or should that fix happen separately?

Reply:
Well, first, let's double check me by confirming what React does. If React doesn't match master, then we probably should get in a PR to fix that prior to this one.

Reply:
It looks like master (and this PR) has the correct behavior. [React first checks if `shouldComponentUpdate` is a function, and _then_ checks if the component is a `PureComponent`](https://github.com/facebook/react/blob/4846809370737f13c0465ac7b83764b408c6e732/packages/react-reconciler/src/ReactFiberClassComponent.js#L259-L284).